### PR TITLE
docs(docs): setup docs search

### LINKED ISSUE #1433: docs: add search support
# Problem Solved By The Feature
When migrating to a Docusaurus website, we lost out-of-the-box search functionality. This increases the burden of finding a specific piece of information on the website and online documentation.

# Solution
## Option A: Algolia DocSearch
The `@docusaurus/preset-classic` preset has [native support for Algolia DocSearch](https://docusaurus.io/docs/search#using-algolia-docsearch) and open-source projects can [apply for a free Algolia DocSearch license](https://docsearch.algolia.com/apply/).

## Option B: Local search
Suggestion: [Docusaurus Search](https://easyops-cn.github.io/docusaurus-search-local/) is a local/offline search plugin for Docusaurus, meaning no 3rd party service is required. See alternatives in *alternatives Considered*.

# alternatives Considered
## Client-side search
[docusaurus-plugin-lunr](https://github.com/daldridge/docusaurus-plugin-lunr) - Docusaurus v2 plugin to create a local search index for use with Lunr.js
[docusaurus-lunr-search](https://github.com/lelouch77/docusaurus-lunr-search) - Offline Search for Docusaurus v2
[docusaurus-search-local](https://github.com/cmfcmf/docusaurus-search-local) - Offline/local search for Docusaurus v2
[docusaurus-theme-search-typesense](https://github.com/typesense/docusaurus-theme-search-typesense) - Docusaurus v2 plugin for [Typesense DocSearch](https://typesense.org/docs/latest/guide/docsearch.html).

# Additional Context

### PR DESCRIPTION
**Pull request by mikelgo**
This PR will close #1433 

I setup search using [docusaurus-search-local](https://github.com/easyops-cn/docusaurus-search-local).

To test it locally you will need to
* `nx run docs:build`
* serve the build docs in the dist folder. Navigate to dist/apps/docs and serve the files e.g. with `npx http-server` (https://github.com/http-party/http-server)

é voilá:
image

In the documentation of the search-plugin there's also an option mentioned on how to have the search available in dev mode  (without serving the build files from a web server) but at least i could not make it work with the nx setup. 

Hint: if you are not serving the build docs from a web server and instead run the project in dev mode with `nx serve docs` you will get a warning when using the search bar:

image

### CODE REVIEW

**Comment by LayZeeDK** [comment]
Thank you, @mikelgo! 🙌 This is awesome!

Please rebase on `main` and try it out with the many documentation additions deployed by #1436.

**Review by LayZeeDK — CHANGES_REQUESTED** [review]
Thank you, @mikelgo! 🙌 This is awesome!

Please rebase on `main` and try it out with the many documentation additions deployed by #1436.

File: apps/docs/package.json [hunk 1]
```diff
@@ -8,11 +8,13 @@
     "clear": "docusaurus clear",
     "write-translations": "docusaurus write-translations",
     "write-heading-ids": "docusaurus write-heading-ids",
-    "typecheck": "tsc"
+    "typecheck": "tsc",
+    "build": "docusaurus build"
```
Comment (LayZeeDK) [inline apps/docs/package.json:12]: issue: We don't need this script since we have `nx build docs`
```suggestion
    "typecheck": "tsc"
```

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

**Comment by LayZeeDK** [comment]
@mikelgo 
We have the `serve-static` target for the `docs` project which calls `docusaurus serve` behind the scene. Local search works when using `nx serve-static docs` 👍

File: apps/docs/docusaurus.config.js [hunk 2]
```diff
@@ -52,6 +52,17 @@ module.exports = {
...
   ],
   tagline: 'Performance & DX',
+  themes: [
+    [
+      require.resolve("@easyops-cn/docusaurus-search-local"),
+      {
+        hashed: true,
+        indexDocs: true,
+        indexPages: true,
+        language: ['en' ],
```
Comment (LayZeeDK) [inline apps/docs/docusaurus.config.js:62]: nit: Leave out the extraneous space.
```suggestion
        language: ['en'],
```

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

**Comment by mikelgo** [comment]
> @mikelgo We have the `serve-static` target for the `docs` project which calls `docusaurus serve` behind the scenes. Local search works when using `nx serve-static docs` 👍

Great, thanks for pointing that out 👍

**Comment by mikelgo** [comment]
@LayZeeDK Thanks for your review, I resolved all the points. 
However I screw up the rebasing with the remote origin, now I have 2 commits.
Originally I wanted to merge the changes from the review in commit c7d9f264c118022904b80f207c3c5271ef3c342f 

maybe you can help me?

**Review by LayZeeDK — APPROVED** [review]
Thank you, @mikelgo. This is a big help to the RxAngular team and the rest of the RxAngular community!

**Comment by mikelgo** [comment]
Thanks for your kind words @LayZeeDK, really appreciate it. 
I hope I can provide even more helpful things in the future :-)

**Pull request merged**